src/burnchains/db.rs

```diff
@@ -337,9 +1080,146 @@ impl BurnchainDB {
         ops
     }
 
-    pub fn store_new_burnchain_block(
+    pub fn get_affirmation_map(
+        conn: &DBConn,
+        affirmation_id: u64,
+    ) -> Result<Option<AffirmationMap>, DBError> {
+        let sql = "SELECT affirmation_map FROM affirmation_maps WHERE affirmation_id = ?1";
+        let args: &[&dyn ToSql] = &[&u64_to_sql(affirmation_id)?];
+        query_row(conn, sql, args)
+    }
+
+    pub fn get_affirmation_weight(
+        conn: &DBConn,
+        affirmation_id: u64,
+    ) -> Result<Option<u64>, DBError> {
+        let sql = "SELECT weight FROM affirmation_maps WHERE affirmation_id = ?1";
+        let args: &[&dyn ToSql] = &[&u64_to_sql(affirmation_id)?];
+        query_row(conn, sql, args)
+    }
+
+    pub fn get_affirmation_map_id(
+        conn: &DBConn,
+        affirmation_map: &AffirmationMap,
+    ) -> Result<Option<u64>, DBError> {
+        let sql = "SELECT affirmation_id FROM affirmation_maps WHERE affirmation_map = ?1";
```

Comment:
Should `affirmation_map` be an indexed column?

Reply:
Yup; done